march_shared_classes/test/subgait_test.py

```diff
@@ -140,3 +153,70 @@ def test_equalize_amount_of_setpoints_with_lower_duration_new_gait(self):
                          msg='Scaling the function did not result in same timestamps and equal amount of setpoints'
                              '\nold timestamps: {old} \nnew timestamps: {new}'
                          .format(old=str(timestamps), new=str(self.subgait.get_unique_timestamps())))
+
+    # Subgait.interpolate_subgaits tests
+    def load_interpolatable_subgaits(self, subgait_name='left_close', base_version='MV_walk_leftclose_v1',
+                                     other_version='MV_walk_leftclose_v2'):
+        base_subgait_path = '{rsc}/{gait}/{subgait}/{version}.subgait'.format(rsc=self.resources_folder,
+                                                                              gait=self.gait_name,
+                                                                              subgait=subgait_name,
+                                                                              version=base_version)
+        base_subgait = Subgait.from_file(self.robot, base_subgait_path)
+        other_subgait_path = '{rsc}/{gait}/{subgait}/{version}.subgait'.format(rsc=self.resources_folder,
+                                                                               gait=self.gait_name,
+                                                                               subgait=subgait_name,
+                                                                               version=other_version)
+        other_subgait = Subgait.from_file(self.robot, other_subgait_path)
+        return base_subgait, other_subgait
+
+    def test_interpolate_subgaits_wrong_parameter(self):
+        # should be 0 <= parameter <= 1
+        base_subgait, other_subgait = self.load_interpolatable_subgaits()
+        with self.assertRaises(ValueError):
+            Subgait.interpolate_subgaits(base_subgait, other_subgait, 2)
+
+    def test_interpolate_subgaits_parameter_zero(self):
+        base_subgait, other_subgait = self.load_interpolatable_subgaits()
+        new_subgait = Subgait.interpolate_subgaits(base_subgait, other_subgait, 0)
+        self.assertEqual(base_subgait, new_subgait)
+
+    def test_interpolate_subgaits_parameter_one(self):
+        base_subgait, other_subgait = self.load_interpolatable_subgaits()
+        new_subgait = Subgait.interpolate_subgaits(base_subgait, other_subgait, 1)
+        self.assertEqual(other_subgait, new_subgait)
+
+    def test_interpolate_subgaits_interpolated(self):
+        # test whether each setpoint of the new subgait is between the setpoins
+        base_subgait, other_subgait = self.load_interpolatable_subgaits()
+        new_subgait = Subgait.interpolate_subgaits(base_subgait, other_subgait, 1)
```

Comment:
```suggestion
        new_subgait = Subgait.interpolate_subgaits(base_subgait, other_subgait, 0.4)
```
You could even go so far as to input a random value between 0 and 1 here

Reply:
There is this packages hypothesis, which can do this very well and searches for edge cases.

Reply:
I agree that 1 is not the best value here, although I think random is not the best practice in testing. We might consider using the hypothesis package if we want to test this more generally.